matisse/src/main/java/com/zhihu/matisse/MimeType.java

```diff
@@ -150,12 +151,22 @@ public boolean checkType(ContentResolver resolver, Uri uri) {
             return false;
         }
         String type = map.getExtensionFromMimeType(resolver.getType(uri));
+        String path = null;
+        // lazy load the path and prevent resolve for multiple times
+        boolean pathParsed = false;
         for (String extension : mExtensions) {
             if (extension.equals(type)) {
                 return true;
             }
-            String path = PhotoMetadataUtils.getPath(resolver, uri);
-            if (path != null && path.toLowerCase(Locale.US).endsWith(extension)) {
+            if (!pathParsed) {
+                // we only resolve the path for one time
+                path = PhotoMetadataUtils.getPath(resolver, uri);
```

Comment:
Because if `extension.equals(type)` we don't need to parse the path. 
If move `getPath()` & `toLowerCase()` to the position above the loop we will at least parse once. We can avoid parse path sometimes if we write like this.